PermissionsScrapper/PermissionsScrapper/PermsDescription.cs

```diff
@@ -0,0 +1,141 @@
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace PermissionsScrapper
+{
+    public static class PermsDescription
+    {
+        static string _scopesDescriptionsJson = null;
+
+        /// <summary>
+        /// Timer function that fetches permissions descriptions from a Service Principal
+        /// and uploads them to a GitHub repo at specified times.
+        /// </summary>
+        /// <param name="myTimer">Trigger function every weekday at 9 AM UTC</param>
+        /// <param name="log"></param>
+        [FunctionName("DescriptionsScrapper")]
+        public static void Run([TimerTrigger("0 0 9 * * 1-5")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                ApplicationConfig config = ApplicationConfig.ReadFromJsonFile("local.settings.json");
+
+                var result = GetAuthentication(config);
+
+                if (result != null)
+                {
+                    var scopesDescriptionsJson = GetScopesDescriptionsJson(config, result);
+
+                    if (!scopesDescriptionsJson.Equals(_scopesDescriptionsJson, StringComparison.OrdinalIgnoreCase))
+                    {
+                        _scopesDescriptionsJson = scopesDescriptionsJson;
+
+                        // TODO: Call GitHub API with json file to upload to devx-content-repo
+                    }
+                }
+                else
+                {
+                    log.LogInformation($"Failed to get authentication at: {DateTime.Now}");
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogInformation($"Exception occurred: {ex.InnerException.Message ?? ex.Message}\r\n Time of occurrence: {DateTime.Now}");
+            }
+        }
+
+        /// <summary>
+        /// Gets authentication to a protected Web API.
+        /// </summary>
+        /// <param name="config">The application configuration settings.</param>
+        /// <returns>An authentiction result, if successful.</returns>
+        private static AuthenticationResult GetAuthentication(ApplicationConfig config)
+        {
+            IConfidentialClientApplication app;
+            app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
+                  .WithClientSecret(config.ClientSecret)
+                  .WithAuthority(new Uri(config.Authority))
+                  .Build();
+
+            string[] scopes = new string[] { $"{config.ApiUrl}.default" };
+
+            try
+            {
+                return app.AcquireTokenForClient(scopes)
+               .ExecuteAsync().GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                throw ex;
```

Comment:
planning to do something with the exception?

Reply:
Removed the exceptions; we are now handling them from the calling function.